river/InputConfig.zig

```diff
@@ -0,0 +1,474 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2021 The River Developers
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const Self = @This();
+
+const build_options = @import("build_options");
+const std = @import("std");
+const wlr = @import("wlroots");
+
+const log = std.log.scoped(.input_config);
+
+const c = @import("c.zig");
+
+const server = &@import("main.zig").server;
+const util = @import("util.zig");
+
+const CommandError = @import("command.zig").Error;
+const InputDevice = @import("InputManager.zig").InputDevice;
+
+// TODO - keyboards
+//      - mapping to output / region
+//      - calibration matrices
+//      - scroll factor
+
+pub const EventState = enum {
+    unset,
+    enabled,
+    disabled,
+    disabled_on_external_mouse,
```

Comment:
If we used `@"disabled-on-external-mouse"` instead the parse function could be replaced by `std.meta.stringToEnum()`, which would be cleaner IMO. The same goes for all the other enums in this file.